sxm.pl

```diff
@@ -1289,60 +1238,125 @@ package main;
                 }
-            } elsif ($path =~ /\.aac$/) {
-                my $segment_path = substr($path, 1); # Remove leading slash
-                my $data = $sxm->get_segment($segment_path);
                 
-                if ($data) {
-                    my $response = HTTP::Response->new(HTTP_OK);
-                    $response->header('Content-Type' => 'audio/x-aac');
-                    $response->content($data);
-                    $connection->send_response($response);
-                } else {
-                    $connection->send_error(HTTP_INTERNAL_SERVER_ERROR);
-                }
-            } elsif ($path =~ /\/key\/1$/) {
-                my $response = HTTP::Response->new(HTTP_OK);
-                $response->header('Content-Type' => 'text/plain');
-                $response->content($HLS_AES_KEY);
-                $connection->send_response($response);
-            } else {
-                # Send a simple HTML page with instructions
-                my $response = HTTP::Response->new(HTTP_OK);
-                $response->header('Content-Type' => 'text/html');
-                my $html = <<EOT;
+                # No more requests from this connection
+                last REQUEST_LOOP unless $request;
+                
+                # Process the request with comprehensive error handling
+                eval {
+                    my $path = $request->uri->path;
+                    
+                    # Minimal request logging to improve performance
+                    if ($debug >= 2) {
+                        printf "Request: %s %s\n", $request->method, $path;
+                        STDOUT->flush();
+                    }
+                    
+                    my $start_time = time();
+                    
+                    if ($path =~ /\.m3u8$/) {
+                        my ($channel_name) = $path =~ m|/([^/]+)\.m3u8$|;
+                        printf "Playlist: %s\n", $channel_name if $debug;
+                        STDOUT->flush() if $debug;
+                        
+                        my $data = $sxm->get_playlist($channel_name);
+                        
+                        if ($data) {
+                            my $response = HTTP::Response->new(HTTP_OK);
+                            $response->header('Content-Type' => 'application/x-mpegURL');
+                            $response->header('Cache-Control' => 'no-cache');
+                            $response->content($data);
+                            
+                            # Safe response sending with error handling
+                            eval { 
+                                $connection->send_response($response); 
+                                $connection->flush() if $connection->can('flush');
+                            };
+                            if ($@) {
+                                warn "Client disconnected during playlist send: $@" if $debug;
+                            }
```

Comment:
Is there anything blocking in this section.  Can buffers be increased so this does not block.  Or possibly write data to the socket in the background.